src/Symfony/Component/Cache/Adapter/ChainAdapter.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Adapter;
+
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\Cache\Exception\InvalidArgumentException;
+
+/**
+ * Chains adapters together.
+ *
+ * Saves, deletes and clears all registered adapter.
+ * Gets data from the first chained adapter having it in cache.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class ChainAdapter implements AdapterInterface
+{
+    private $adapters = array();
+
+    /**
+     * @param AdapterInterface[] $adapters
+     */
+    public function __construct(array $adapters)
+    {
+        if (2 > count($adapters)) {
```

Comment:
Why? When making this configurable by the user, you might end up with just one adapter, which is fine. I would change the check to `1 < count()`.